lib/metricAggregators.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const util = require('./util');
+
+/**
+ * Returns a new function that applies the `aggregatorFn` to the values.
+ * @param {Function} aggregatorFn function to apply to values.
+ * @return {Function} aggregator function
+ */
+const AggregatorFactory = function(aggregatorFn) {
+	return metrics => {
+		if (metrics.length === 0) return;
+		const result = {
+			help: metrics[0].help,
+			name: metrics[0].name,
+			type: metrics[0].type,
+			values: []
+		};
+		// Gather metrics by labels.
+		const byLabels = new Map();
+		metrics.forEach(metric => {
```

Comment:
this part seems duplicated from `aggregate`. Can it be put in `utils`? (I realize the diff is `.name` vs `.labels`)